wiki/osu!_File_Formats/Osu_(file_format)/en.md

```diff
@@ -194,26 +194,58 @@ Hit Circle Syntax:
 `x,y,time,type,hitSound,addition`
 `164,260,2434,1,0,0:0:0:0:`
 
+x (Integer) ranges from 0 to 512 (inclusive) and y (Integer) ranges from 0 to 384 (inclusive).
+
+time (Integer) is in milliseconds from the beginning of the song.
+
+type (Integer) is a bitmap:
+
+1 = Circle, 2 = Slider, 4 = New combo, 8 = Spinner. More significant bits can be used to choose the next combo color.
+
+hitSound (Integer) is a bitmap:
+
+2 = Whistle, 4 = Finish, 8 = Clap
+
+The only actual "types" are circles, sliders, and spinners. New combo can be OR'd in to extend the behavior of the hit circle.
+
+addition (Integer:Integer:Integer:Integer) is optional, and defaults to "0:0:0:0:". The first value corresponds to the sample set, and the second value corresponds to additions.
+
+0 = Auto, 1 = Normal, 2 = Soft, 3 = Drum
+
+The third value corresponds to the sample index, e.g. normal-3. The fourth value declares the volume of the sample.
+
 Slider Syntax:
 
-`x,y,time,type,hitSound,sliderType|curveX:curveY|...|repeat,pixelLength|edgeHitsound:edgeAddition,addition`
+`x,y,time,type,hitSound,sliderType|curvePoints,repeat,pixelLength,edgeHitsounds,edgeAdditions,addition`
 `424,96,66,2,0,B|380:120|332:96|332:96|304:124,1,130,2|0,0:0|0:0,0:0:0:0:`
 
+x, y, time, and type behave the same as described in Hit Circle Syntax.
+
+hitSound and addition apply to all circles of the slider.
```

Comment:
the slider hitsound value itself only changes the slider body hitsound--- as far as I know, the only extra hitsound that actually does anything here is whistle, since it'll play a continuous whistle as you follow the slider

however, when you change the slider's (overall) hitsound in-editor, it changes more than just this value, it'll also change all of the edge hitsounds to include the new global one. let's say you had a slider with hitsound `0` and edges `0|0|8`. changing the overall hitsound to "whistle" in-editor would result in hitsound `2` and edges `2|2|10`.

the editor makes things a bit more confusing here, but basically all of these values are totally independent from each other. none depend on the others.